regcore/db/es.py

```diff
@@ -66,23 +66,27 @@ def listing(self, label=None):
 
 class ESLayers(ESBase, interface.Layers):
     """Implementation of Elastic Search as layers backend"""
-    def _transform(self, layer, layer_name):
+    def _transform(self, layer, layer_name, doc_type):
         """Add some meta data fields which are ES specific"""
         layer = dict(layer)     # copy
-        reference = layer['reference']
-        del layer['reference']
-        return {'id': '{}:{}'.format(layer_name, reference), 'layer': layer}
+        doc_id = self.sanitize_doc_id(layer['doc_id'])
+        del layer['doc_id']
+        return {'id': ':'.join(layer_name, doc_type, doc_id), 'layer': layer}
 
-    def bulk_put(self, layers, layer_name, prefix):
+    def bulk_put(self, layers, layer_name, doc_type, root_doc_id):
         """Store all layer objects. Note this does not delete existing docs;
         it only replaces/inserts docs, which has loop holes"""
         self.es.bulk_index(
             settings.ELASTIC_SEARCH_INDEX, 'layer',
-            [self._transform(l, layer_name) for l in layers])
+            [self._transform(l, layer_name, doc_type) for l in layers])
 
-    def get(self, name, reference):
+    def sanitize_doc_id(self, doc_id):
```

Comment:
Pedantic note: this method is effectively static. Does it need to be part of a class?